stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
-from _typeshed import Incomplete
+from collections.abc import Callable, Generator, Iterator, Sequence
+from re import Match, Pattern
+from typing import Any, Generic, TypeVar, overload
+from typing_extensions import Self, TypeAlias
 
-def __getattr__(name: str) -> Incomplete: ...
+_TransitionResult: TypeAlias = tuple[list[str], str | None, list[str]]
+_TransitionMethod: TypeAlias = Callable[[Match[str], list[str], str], _TransitionResult]
+
+class StateMachine:
+    input_lines: StringList | None
+    input_offset: int
+    line: str | None
+    line_offset: int
+    debug: bool
+    initial_state: str
+    current_state: list[State]
+    states: dict[str, State]
+    observers: list[Callable[[Self], None]]
```

Comment:
oh confusing.

so based on your description it's not actually `Callable[[str | None, int | None], None]`

it's

`Callable[[str, int], None]` or
`Callable[[str, None], None]` or
`Callable[[None, None], None]` but *not*
`Callable[[None, int], None]`

@JelleZijlstra is there a compact way to annotate that?

Reply:
Actually your callable should support those kind of inputs except for the combination (None, int), but I don't think we can type it like that unless we have an intersection type. So we could technically allow `Callable[[str | None, int | None], None]` even though it would never be called with (None, int) :/

Reply:
It's actually incorrect because it means that a function that only accept None and None is allowed but then it cannot process the normal inputs being str and int.